First commit

P1antain opened the pull request:
For checking

P1antain committed:
First commit

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--10.71--> Проект соответствует сгенерированному под него макету.
`Имеется лишняя кнопка. Цвет хедера не соответствует макету.`
<img width="176" alt="Снимок экрана 2021-08-12 в 12 44 25" src="https://user-images.githubusercontent.com/25247802/129176446-fe6d2a43-7190-4731-9690-595e2dc6e7da.png">
<img width="1242" alt="Снимок экрана 2021-08-12 в 12 42 52" src="https://user-images.githubusercontent.com/25247802/129176488-0dea38db-4abf-4f24-aa4f-bebf1d28bdc9.png">

- [x] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [ ] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`При изменении расширения в карточках остается много лишнего пространства, которое должно быть заполнено контентом ( изображением и текстом) `
<img width="637" alt="Снимок экрана 2021-08-12 в 12 44 55" src="https://user-images.githubusercontent.com/25247802/129192813-7c6193b4-2803-49fe-a0dc-f48524874953.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:x
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Форма поиска фильмов`
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
`У формы поиска фильмов нет свойства required`

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
`Каждый атрибут alt должен быть заполнен`
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 73

P1antain committed:
Fixed error
1) Remove extra btn, add useEffect(temporary). Changed the color of the block header
2) Modified width MovieCard.css
3) Add alt img
4) Add focus element input

coolswood commented:
# Резюме по работе:
### Критерии, влияющие на работоспособность
- [x] <!--10.71--> Проект адаптирован под различные разрешения экрана. Горизонтальный скролл не возникает на разрешениях от 320 пикселей и больше. Нельзя скрывать полосу прокрутки свойством overflow: hidden.
- [x] <!--10.71--> Проект соответствует сгенерированному под него макету.
`Имеется лишняя кнопка. Цвет хедера не соответствует макету.`
<img width="176" alt="Снимок экрана 2021-08-12 в 12 44 25" src="https://user-images.githubusercontent.com/25247802/129176446-fe6d2a43-7190-4731-9690-595e2dc6e7da.png">
<img width="1242" alt="Снимок экрана 2021-08-12 в 12 42 52" src="https://user-images.githubusercontent.com/25247802/129176488-0dea38db-4abf-4f24-aa4f-bebf1d28bdc9.png">

- [x] <!--10.71--> Все блоки из макета свёрстаны. Корректно работает навигация между страницами и ссылки на внешние ресурсы: ни одна ссылка не ведёт в пустоту или на якорь, внешние ссылки открываются в новой вкладке. На все страницы проекта можно попасть, а скрытые блоки можно отобразить.
- [x] <!--10.71--> Отзывчивая вёрстка, которая корректно тянется на всех промежуточных разрешениях.
`При изменении расширения в карточках остается много лишнего пространства, которое должно быть заполнено контентом ( изображением и текстом) `
<img width="637" alt="Снимок экрана 2021-08-12 в 12 44 55" src="https://user-images.githubusercontent.com/25247802/129192813-7c6193b4-2803-49fe-a0dc-f48524874953.png">

- [x] <!--2.86--> Отсутствуют ошибки валидации при сборке приложения или в валидаторе.
- [x] <!--2.86--> В коде используется семантическая разметка: применяются семантические теги, выбор элементов при вёрстке корректен (параграф должен быть параграфом, список — списком); структура DOM-дерева состоит не только из контейнеров div.
- [x] <!--2.86--> Для позиционирования элементов выбран верный подход, описанный корректным синтаксисом.
- [x] <!--2.86--> Каркас макета реализован на Flex layout или Grid layout;
- [x] <!--2.86--> Инфраструктурные файлы проекта созданы через CRA.
- [x] <!--2.86--> Работа модальных окон настроена.
- [x] <!--2.86--> Разметка портирована в JSX:x
- [x] <!--2.86--> разметка заключена в ( ), лишние пустые <div> для этого не используются;
- [x] <!--2.86--> вынесена в соответствующие ей компоненты.
- [x] <!--2.86--> В проекте есть:
- [x] <!--2.86--> директория images/ с картинками,
- [x] <!--2.86--> директория components/ с JS и CSS-файлами компонентов,
- [x] <!--2.86--> директория fonts/ со шрифтами.

### Хорошие практики
- [x] <!--2.14--> Именование классов сделано по БЭМ.
- [x] <!--2.14--> Кнопки, инпуты и ссылки реализованы во всех состояниях, указанных в макете.
- [x] <!--2.14--> В макете есть одинаковые элементы. Для их оформления должен быть переиспользован один компонент.
- [x] <!--2.14--> Шрифты подключены через @font-face.
- [x] <!--2.14--> reset.css запрещен;
- [ ] <!--2.14--> Элементы формы должны выделяться, когда на них установлен фокус. Состояние фокуса, а также outline стилизованы в соответствии с макетом или на своё усмотрение.
`Форма поиска фильмов`
- [ ] <!--2.14--> У формы должны быть заданы плейсхолдеры, валидированы обязательные поля.
`У формы поиска фильмов нет свойства required`

### Рекомендации
- [x] <!--1.67--> Для изображений задан атрибут alt с подходящий значением.
`Каждый атрибут alt должен быть заполнен`
- [x] <!--1.67--> Для каждого шрифта указаны альтернативные варианты из системных шрифтов.
- [x] <!--1.67--> Растровые и векторные изображения оптимизированы;

## Количество баллов: 96
